### PR TITLE
p2p: Log addresses of stalling peers

### PR DESCRIPTION
**Pull request by mzumsande**
This was suggested in #27705 by ArmchairCryptologist.
It allows node operators that have the `-logips` option enabled to better identify potentially misbehaving peers and maybe ban them.
This is especially helpful in case of inbound peers for which (dis)connections aren't logged per default, so it's impossible to use the debug log to connect their `nodeId` to an address unless the very noisy `net` debugging is enabled.
In case of outbound peers for which the address is potentially logged when establishing the connection, this just adds some convenience.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -5751,7 +5751,7 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
             // Stalling only triggers when the block download window cannot move. During normal steady state,
             // the download window should be much larger than the to-be-downloaded set of blocks, so disconnection
             // should only happen during initial block download.
-            LogPrintf("Peer=%d is stalling block download, disconnecting\n", pto->GetId());
+            LogPrintf("Peer=%d%d is stalling block download, disconnecting\n", pto->GetId(), fLogIPs ? strprintf(" addr=%s", pto->addr.ToStringAddrPort()) : "");
```
Comment (ajtowns) [inline src/net_processing.cpp:5754]: Should be `Peer=%d%s` (the result of strprintf isn't a number) and `peeraddr=%s` (for better consistency with other `fLogIPs` logs)

Comment (mzumsande) [reply src/net_processing.cpp]: fixed, thanks.

**Review by 0xB10C — APPROVED** [review]
Untested ACK fb02a3cd1a105bdf60ca39e1858e77685be88976

Somewhat related: net logging for specific IP subnets/addresses could also be useful for debugging something like #27705. Probably a bigger change though.

**Review by instagibbs — APPROVED** [review]
utACK fb02a3cd1a105bdf60ca39e1858e77685be88976

**Comment by jamesob** [comment]
github ACK https://github.com/bitcoin/bitcoin/commit/fb02a3cd1a105bdf60ca39e1858e77685be88976

**Comment by Sjors** [comment]
May I suggest a fine pairing? #27826

**Comment by mzumsande** [comment]
Yes, I'll review soon! It's been on my list anyway but I got distracted by other things...

**Pull request merged**